### PR TITLE
feat: use human readable title instead of field title

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR modifies the behaviour for dgs-v2 pages. For datasets on dgs v2, fields have both an internal name (e.g. `notification_no`) and a human readable name (e.g. `Notification number`). We previously were using the internal name for the display of the table header, as well as the field dropdown, as the dgs-v1 datasets only contain an internal name. This PR changes that behaviour such that dgs-v2 pages will use the new human readable names for page display, while still sending the internal name to be used for search queries.

Note that this means the field provided by the frontmatter should now be the human readable name. However, existing pages are unaffected in functionality - for the 2 sites currently making use of dgs-v2 pages, the deslugified versions of the internal representation and the field name are identical.

### CODE REVIEW

File: assets/js/datagovsg-search.js [hunk 1]
```diff
@@ -73,18 +73,29 @@ function databaseSearch(searchTerm, index, callback) {
...
 
-  let request
+  // columnMetadata:
+  //   id: represents searchable field
+  //   title: represents human readable field
+  let request, columnMetadata
   let formattedSearchField = searchField ? searchField.replace(" ", "_") : ""
-  if (!!searchField) {
-    // Datagov-v2 search
+  if (!!defaultField) {
```
Comment (seaerchin) [inline assets/js/datagovsg-search.js]: why was this change required?

File: assets/js/datagovsg-search.js [hunk 2]
```diff
@@ -106,6 +117,14 @@ function databaseSearch(searchTerm, index, callback) {
...
   request.done(function (data) {
     datagovsgTotal = data.result.total;
+    if (!columnMetadata) {
+      // V1 search, id and title are the same
+      columnMetadata = data.result.fields.map(field => ({
+        id: field.id,
+        title: field.id,
+      }))
+    }
+    columnMetadata = columnMetadata || data.result.fields
```
Comment (seaerchin) [inline assets/js/datagovsg-search.js]: this is confusing on 2 fronts: 
1. `columnMetadata` is defined up above but we do the mapping here
2. why do the `map` if `data.result.fields` is sufficient? 

without context, i'd have thought `data.result.fields` alone would be enough. also, `columnMeta` exists only if there's a default field - i'm assuming without, we will use v1?

also, this is abit confusing here - we initialise`columMetadata` on `resp.data` - we seem to be assuming order of operations here (that the earlier initialisation will occur **before** `request.done`) but i'm not entirely sure this is true, especially in an async context.

can we just verify this behaviour or split up `columnMetadata` by the `if/else`?

File: assets/js/datagovsg-search.js [hunk 3]
```diff
@@ -115,12 +134,12 @@ function databaseSearch(searchTerm, index, callback) {
 
     // The fieldArray is the array containing the field names in the data.gov.sg table
     const removableFields = ["_id", "_full_count", "rank", `rank ${formattedSearchField}`]
-    fieldArray = remove(data.result.fields, removableFields);
+    fieldArray = remove(columnMetadata, removableFields);
```
Comment (seaerchin) [inline assets/js/datagovsg-search.js:136]: pointing htis out because of sussy stuff wrt `columMetadata` above

**Review by seaerchin — COMMENTED** [review]
we should aim to verify behaviour of async code here, it's potentially buggy. 

style-wise, the disjoint between initialisation and assignment is abit confusing

File: assets/js/datagovsg-search.js (hunk 1 above)
Comment (alexanderleegs) [reply assets/js/datagovsg-search.js]: It was previously incorrect and only applying for dgs-v2 layouts where someone had entered a search term - it was supposed to be for any dgs-v2 layout

File: assets/js/datagovsg-search.js (hunk 2 above)
Comment (alexanderleegs) [reply assets/js/datagovsg-search.js]: sorry the `columnMetadata = columnMetadata || data.result.fields` assignment isn't useful - have removed it in https://github.com/isomerpages/isomerpages-template/pull/377/commits/282e2fa857051007c5153b2086a3899130e906d8

The order of operations here holds because this is only executed after `request.done` - where the earlier assignment is done

**Review by seaerchin — APPROVED** [review]
lgtm pls add comment for the order cos it mgiht be confusing
